Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -566,6 +567,45 @@ public class PBXProjGenerator {
         let childGroups = group.children.compactMap { $0 as? PBXGroup }
         childGroups.forEach(sortGroups)
     }
+    
+    func setupGroupOrdering(group: PBXGroup) {
+        let groupOrdering = project.options.groupOrdering.first { groupOrdering in
+            let groupName = group.nameOrPath
+            
+            if groupName == groupOrdering.pattern {
+                return true
+            }
+            
+            if let regex = groupOrdering.regex {
+                return regex.isMatch(to: groupName)
+            }
+            
+            return false
+        }
+        
+        if let order = groupOrdering?.order, let fileSortPosition = groupOrdering?.fileSortPosition {
+            let files = group.children.filter { $0 is PBXFileReference }
+            var groups = group.children.filter { $0 is PBXGroup }
+            
+            var filteredGroups = [PBXFileElement]()
+            
+            for groupName in order {
+                guard let group = groups.first(where: { $0.nameOrPath == groupName }) else {
+                    continue
+                }
+                
+                filteredGroups.append(group)
+                groups.removeAll { $0 == group }
+            }
```

Comment:
I think this will cause performance issues in large projects that use a regex pattern, as there are a lot of nested loops here, but I guess it's up to the project author to take on that cost